.github/workflows/build_and_test_snap.yaml

```diff
@@ -0,0 +1,67 @@
+# Note: The trigger for GPU tests is based off the name of this workflow.
+#       If we rename this workflow, update the GPU test trigger as well.
+name: Build and test Snap
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+env:
+  SNAP_ARTIFACT_NAME: dss-snap
+  SNAP_FILE: dss.snap
+  KUBECONFIG: ~/.dss/config
+
+jobs:
+  build-snap:
+    name: Build the snap
+    runs-on: ubuntu-22.04
+    outputs:
+      snap-file: ${{ steps.snapcraft.outputs.snap }}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - uses: snapcore/action-build@v1.2.0
+      id: snapcraft
+      with:
+        # Use a deterministic file name, so we can easily reference it later
+        snapcraft-args: "-o ${{ env.SNAP_FILE }}"
+
+    - uses: actions/upload-artifact@v4
+      id: upload-snap-artifact
+      with:
+        name: ${{ env.SNAP_ARTIFACT_NAME }}
+        path: ${{ steps.snapcraft.outputs.snap }}
+
+  test:
+    name: Test the snap
+    runs-on: ubuntu-22.04
+    needs:
+      - build-snap
+    steps:
+    - uses: actions/download-artifact@v4
+      id: download-snap-artifact
+      with:
+        name: ${{ env.SNAP_ARTIFACT_NAME }}
+
+    - name: Install the snap locally
+      # use --dangerous so we trust the snap even though it doesn't have signatures
+      run: |
+        sudo snap install ${{ env.SNAP_FILE }} --dangerous
+        sudo snap connect data-science-stack:dot-dss-config
+
+    - name: Set up Microk8s
+      uses: balchua/microk8s-actions@v0.3.2
+      with:
+        channel: '1.28/stable'
+        addons: '["hostpath-storage"]'
+
+    - name: Export kubeconfig where strict snap can access it
+      run: |
+        mkdir -p ~/.dss
+        cp ~/.kube/config ${{ env.KUBECONFIG }}
+
+    - name: Test dss snap
+      run: |
+        data-science-stack.dss initialize --kubeconfig ${{ env.KUBECONFIG }}
```

Comment:
Lets create a tox environment for this so SolQA can just execute it on their machines. I also think we should test all the commands. I propose:
- initialize 
- run status
- create notebook
- list notebooks 
- stop notebook 
- start notebook 
- remove notebook 
- purge DSS

Reply:
I've added these, but I don't think we should include so much for two reasons:
* The proposed list of tests is not idempotent.  This test does not run isolated, it runs in a global microk8s.  If `dss list` fails, the test will leave the sample notebook in the cluster.  Running the test a second time will then fail because the first test didn't clean itself up (sample notebook already exists).  
*   Everything listed is tested in the `integration` environment.  What we're testing here is stuff that packaging it into a snap can break.  

imo what we need here is just:
* `dss status`

That command needs a valid kubeconfig in an accessible location, will break if there's no microk8s access, and is idempotent so we don't need to worry about it leaving fragments around when tests fail.  wdyt?

Reply:
After some thinking ... your afrgument makes sense ... lets just run initialize and status.... for the gpu I will create separate python librarry integration test for deploying gpu notebook and making sure it works.

Reply:
@misohu and I spoke offline and are reducing this to just run `dss status` since it should fully test the snap.  But for the gpu tests added later, they'll need more.